modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/migration/MigrationFromMongo.java

```diff
@@ -0,0 +1,546 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.console.migration;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Filters;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.console.dto.Account;
+import org.apache.ignite.console.dto.Activity;
+import org.apache.ignite.console.dto.ActivityKey;
+import org.apache.ignite.console.dto.Notebook;
+import org.apache.ignite.console.json.JsonArray;
+import org.apache.ignite.console.json.JsonObject;
+import org.apache.ignite.console.repositories.AccountsRepository;
+import org.apache.ignite.console.repositories.ConfigurationsRepository;
+import org.apache.ignite.console.repositories.NotebooksRepository;
+import org.apache.ignite.console.tx.TransactionManager;
+import org.apache.ignite.console.web.model.ConfigurationKey;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.transactions.Transaction;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import static org.apache.ignite.console.migration.MigrateUtils.asPrimitives;
+import static org.apache.ignite.console.migration.MigrateUtils.asStrings;
+import static org.apache.ignite.console.migration.MigrateUtils.getDocument;
+import static org.apache.ignite.console.migration.MigrateUtils.getInteger;
+import static org.apache.ignite.console.migration.MigrateUtils.mongoIdsToNewIds;
+import static org.apache.ignite.console.migration.MigrateUtils.mongoToJson;
+import static org.apache.ignite.console.migration.MigrateUtils.off;
+import static org.apache.ignite.console.utils.Utils.fromJson;
+
+/**
+ * Service to migrate user data from MongoDB to GridGain persistence.
+ */
+@Service
+public class MigrationFromMongo {
+    /** */
+    private static final Logger log = LoggerFactory.getLogger(MigrationFromMongo.class);
+
+    /** */
+    private final TransactionManager txMgr;
+
+    /** */
+    private final AccountsRepository accRepo;
+
+    /** */
+    private final NotebooksRepository notebooksRepo;
+
+    /** */
+    private final ConfigurationsRepository cfgsRepo;
+
+    /** */
+    @Qualifier("migration")
+    private final MigrateActivitiesRepository activitiesRepo;
+
+    /** */
+    protected MongoDatabase mongoDb;
+
+    /** */
+    @Value("${migration.enabled:false}")
+    private boolean enabled;
```

Comment:
Why we need this flag? Check mongo url and that database is empty

Reply:
fixed